src/utils.rs

```diff
@@ -4,5 +4,11 @@ use numeric_literals::replace_numeric_literals;
 #[inline(always)]
 #[replace_numeric_literals(T::from(literal).unwrap())]
 pub fn floats_almost_equal<T: Float>(f1: T, f2: T) -> bool {
-    f1 + 1e-7 > f2 && f1 - 1e-7 < f2
```

Comment:
Why the change here? It was valid before

Reply:
We are checking if 2 floats are almost equal, up to floating point error

if `f1 - f2 < 1 e-7`, `f1` could be `0` and `f2` `300000`. The original implementation puts the difference between `f1 - f2` in the interval `{-1e-7; 1e-7}`

Reply:
My bad, i thought this did the same and was easier to read